thesis-api-v2/src/main/java/pl/edu/agh/quizzesthesis/business/service/WrongAnswerService.java

```diff
@@ -94,15 +88,21 @@ private List<String> getWrongAnswers(Term rightAnswerTerm) {
         }
 
         if (wrongAnswers.size() < 3) {
-            for (var potentialAnswer : relatedNounsSorted) {
+            for (var potentialAnswer : relatedNouns) {
                 if (wrongAnswers.size() == 3) {
                     break;
                 }
-                if (potentialAnswer.frequency() > minimumFrequencyThreshold && !wrongAnswers.contains(potentialAnswer.word())) {
+                if (!wrongAnswers.contains(potentialAnswer.word())) {
                     wrongAnswers.add(potentialAnswer.word());
                 }
             }
         }
+        while (wrongAnswers.size() < 3) {
+            var potentialAnswer = termService.getRandom(rightAnswerTerm.getSearchPhrase().getCategory().getId());
```

Comment:
if `termService` stops working properly (for example if it always returns the same term) we might get into infinite loop. I don't really think this is an issue since we do know that `termService` is working, but I just wanted to point out that there is a potential threat and normally it should be fixed but since our application has already a lot of other threats this one can just join a collection ;)

Reply:
I was thinking about something like exiting loop after 10 iterations, but it wouldn't look good in the code

Reply:
I agree, generally speaking the best sollution whould be to have tests that detects such issues, but since we do not have any we can't do much about it.